src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -100,14 +100,41 @@ public void resetOnPath(String path) {
         // Intentionally blank
     }
 
+    /**
+     * Copies an existing file (if not a temporary file) to a target location. Returns the public path for the resulting file.
+     *
+     * @param oldFilePath  the old file path
+     * @param targetFolder the folder that a file should be copied to
+     * @param entityId     id of the entity this file belongs to (needed to generate public path). If this is null, a placeholder will be inserted where the id would be
+     * @return the resulting public path
+     */
+    public String copyExistingFileToTarget(String oldFilePath, String targetFolder, Long entityId) {
```

Comment:
Do we need a new method for this, or could you just copy the file to the temp directory and have `manageFilesForUpdatedFilePath()` take care of it?
It seems like the method body is [copied from below](https://github.com/ls1intum/Artemis/blob/01a79e3e2e8d6c9a16d30b691aa9e370b446f96b/src/main/java/de/tum/in/www1/artemis/service/FileService.java#L180-L194), right?

Reply:
> could you just copy the file to the temp directory and have manageFilesForUpdatedFilePath() take care of it

That is indeed what the method is doing. It copies the file into a temp location using the placeholder path, and then the @PrePersist on DragAndDropQuestion.java takes care of the rest using `manageFilesForUpdatedFilePath()`. The body is copied from below with the important and required difference that we call Files.copy() instead of Files.move().

I think it's better to handle it like this with a specific method instead of changing the `manageFilesForUpdatedFilePath()` to handle yet another use case.

Reply:
I think it would be a great idea to reuse/streamline the file copy functionality for both cases (quiz import and lecture import).